src/index.ts

```diff
@@ -1,46 +1,43 @@
-import { Response, Headers as FetchHeaders } from 'node-fetch';
-import FormData from 'form-data';
+import { Response, Request, Headers as FetchHeaders, RequestInfo, RequestInit } from 'node-fetch';
 import { AxiosInstance, AxiosRequestConfig } from './types';
+import FormData from 'form-data';
 
-export interface FetchInit extends Record<string, any> {
-  headers?: Record<string, string>;
-  method?: AxiosRequestConfig['method'];
-  body?: FormData | any;
-  extra?: any;
-}
-
-export type AxiosTransformer = (config: AxiosRequestConfig, input: string | undefined, init: FetchInit) => AxiosRequestConfig;
+export type AxiosTransformer = (config: AxiosRequestConfig, input: RequestInfo, init?: RequestInit) => AxiosRequestConfig;
 
-export type AxiosFetch = (input?: string, init?: FetchInit) => Promise<Response>;
+export type AxiosFetch = (input: RequestInfo, init?: RequestInit) => Promise<Response>;
 
 /**
  * A Fetch WebAPI implementation based on the Axios client
+ *
+ * @param axios
+ * @param transformer Convert the `fetch` style arguments into a Axios style config
+ * @param input
+ * @param init
  */
 async function axiosFetch (
   axios: AxiosInstance,
-  // Convert the `fetch` style arguments into a Axios style config
-  transformer?: AxiosTransformer,
```

Comment:
Is it possible to leave the optionality of each argument unchanged? I'm hoping to avoid any breaking change

Reply:
Since `axiosFetch` is private, it's a minor point. I like you analysis